localization/pose_estimator/ndt_scan_matcher/ndt_omp/CMakeLists.txt

```diff
@@ -1,70 +1,81 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(ndt_omp)
 
-# -mavx causes a lot of errors!!
-add_definitions(-std=c++14 -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2)
-set(CMAKE_CXX_FLAGS "-std=c++14 -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2")
-
-# warn performance issue of ndt_omp which is NOT built in release mode
-string(TOUPPER ${CMAKE_BUILD_TYPE} uppercase_CMAKE_BUILD_TYPE)
-if (NOT (${uppercase_CMAKE_BUILD_TYPE} STREQUAL "RELEASE"))
-  message(WARNING
-    "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} may cause a serious performance degradation. Please configure with -DCMAKE_BUILD_TYPE=Release.")
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
 endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  roscpp
-  pcl_ros
-)
+set(CMAKE_BUILD_TYPE "Release")
 
+find_package(ament_cmake REQUIRED)
+find_package(PCL REQUIRED COMPONENTS common filters kdtree registration io)
 find_package(OpenMP)
+
 if (OPENMP_FOUND)
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
```

Comment:
I'm surprised that it works w/o updating linker flags as in here. Note that I wrote this 3 years ago

https://github.com/mtazzari/galario/blob/master/src/CMakeLists.txt#L63

Reply:
Changed to a more modern style of using OpenMP.